spoc/query_engine.py

```diff
@@ -172,83 +218,375 @@ def __call__(self, genomic_data: GenomicData) -> GenomicData:
         regions = self._convert_to_duckdb(self._regions)
         # get position columns and construct filter
         position_fields = input_schema.get_position_fields()
+        # add end position if not present
+        if len(position_fields[1]) == 2:
+            genomic_df = self._add_end_position(
+                genomic_df, input_schema.get_binsize(), position_fields
+            )
+            position_fields = {
+                position: [f"chrom_{position}", f"start_{position}", f"end_{position}"]
+                for position in position_fields.keys()
+            }
         # construct query
-        df = genomic_df.set_alias("data").join(
+        snipped_df = genomic_df.set_alias("data").join(
             regions.set_alias("regions"), self._contstruct_filter(position_fields)
         )
-        return QueryResult(df, self._get_transformed_schema(input_schema))
+        return QueryPlan(
+            snipped_df,
+            self._get_transformed_schema(snipped_df, input_schema, position_fields),
+        )
+
 
+class AggregationFunction(Enum):
+    """Enum for aggregation functions.
+    Options are:
+        SUM: Sum of values.
+        AVG_WITH_EMPTY: Average of values, empty values are counted as 0.
+        AVG: Average of values, empty values are not counted.
+        COUNT: Number of values.
+    """
 
-class MappedRegionFilter:
-    """Filter for filtering mapped regions
-    of contacts or pixels."""
+    SUM: str = "SUM"
+    AVG_WITH_EMPTY: str = "AVG_WITH_EMPTY"
```

Comment:
Hmm, I wouldn't do that since `AVG_WITH_EMPTU` behaves different from `nanmean` the former fills in the empty values, whereas the latter ignores them.

Reply:
True!